numba_dpex/experimental/launcher.py

```diff
@@ -28,153 +27,49 @@
 from numba_dpex.core.utils import kernel_launcher as kl
 from numba_dpex.dpctl_iface import libsyclinterface_bindings as sycl
 from numba_dpex.dpctl_iface.wrappers import wrap_event_reference
-from numba_dpex.experimental.kernel_dispatcher import _KernelModule
-from numba_dpex.utils import create_null_ptr
-
-_KernelArgs = namedtuple(
-    "_KernelArgs",
-    [
-        "flattened_args_count",
-        "array_of_kernel_args",
-        "array_of_kernel_arg_types",
-    ],
-)
-
-_KernelSubmissionArgs = namedtuple(
-    "_KernelSubmissionArgs",
-    [
-        "kernel_ref",
-        "queue_ref",
-        "kernel_args",
-        "global_range_extents",
-        "local_range_extents",
-    ],
-)
-
-_LLVMIRValuesForIndexSpace = namedtuple(
-    "_LLVMIRValuesForNdRange", ["global_range_extents", "local_range_extents"]
+from numba_dpex.experimental.kernel_dispatcher import (
+    KernelDispatcher,
+    _KernelModule,
 )
+from numba_dpex.utils import create_null_ptr
 
 
-class _LaunchTrampolineFunctionBodyGenerator:
+class _LaunchKernelIRGenerator:
```

Comment:
The class name is not clear. The previous name was specific to the class being the helper to build the `launch_trampoline` intrinsic function's body.

How about: _SubmitKernelCo

Reply:
I'm actually thinking of getting rid of this class and use functions instead. What do you think? I'm thinking about it, since class is used as builder and then get's destroyed within one function body.

Reply:
No strong opinion, tbh. We will need to pass two extra args to each function and removing the class kind of looses the logical relation between the functions that they all are doing something related. But, if you want to change it go ahead.